http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java

```diff
@@ -213,21 +213,22 @@ private <T> Optional<T> convertByteBuf(Argument<T> type) {
             }
             return Optional.empty();
         }
+        // All content operation should call slice to prevent reading the buffer completely
```

Comment:
Why? Ownership should now be in the response

Reply:
One reader would read the buffer completely and prevent others reading it.

Reply:
We should not read the same body twice.

Reply:
Why not? The body is already buffered in the client response

Reply:
I don't want to keep it that way forever

Reply:
Then we need to find a way around it considering `FullNettyClientHttpResponse` is now reading the body with the predefined type in the constructor which will makes the buffer fully read. Different combinations now make interactions very unpleasant, sometimes the body is there sometimes it's not.

We can introduce something like `bufferEntity` in JAX-RS. I don't really want to dig more into it.

Reply:
ok let's leave it. i think it will need a bigger redesign eventually.